contrib/platform/src/com/sun/jna/platform/win32/Winspool.java

```diff
@@ -565,6 +633,313 @@ boolean OpenPrinter(
+         * a job or printer information field of interest.
+         */
+        public Pointer pFields;
+
+        public void setFields(short[] fields) {
+            final long shortSizeInBytes = 2L;
+            Memory fieldsMemory = new Memory(fields.length * shortSizeInBytes);
+            fieldsMemory.write(0, fields, 0, fields.length);
+            pFields = fieldsMemory;
+            Count = fields.length;
+        }
+
+        public short[] getFields() {
+            return pFields.getShortArray(0, Count);
+        }
+    }
+
+    /**
+     * The PRINTER_NOTIFY_INFO structure contains printer information returned
+     * by the FindNextPrinterChangeNotification function. The function returns
+     * this information after a wait operation on a printer change notification
+     * object has been satisfied.
+     *
+     * @see
+     * <a href="https://docs.microsoft.com/en-us/windows/win32/printdocs/printer-notify-info">
+     *     PRINTER_NOTIFY_INFO structure
+     * </a>
+     */
+    @Structure.FieldOrder({ "Version", "Flags", "Count", "aData" })
+    public class PRINTER_NOTIFY_INFO extends Structure {
+
+        /**
+         * The version of this structure. Set this member to 2.
+         */
+        public int Version;
+
+        /**
+         * A bit flag that indicates the state of the notification structure. If
+         * the PRINTER_NOTIFY_INFO_DISCARDED bit is set, it indicates that some
+         * notifications had to be discarded.
+         */
+        public int Flags;
+
+        /**
+         * The number of PRINTER_NOTIFY_INFO_DATA elements in the aData array.
+         */
+        public int Count;
+
+        /**
+         * An array of PRINTER_NOTIFY_INFO_DATA structures. Each element of the
+         * array identifies a single job or printer information field, and
+         * provides the current data for that field.
+         */
+        public PRINTER_NOTIFY_INFO_DATA[] aData =
+            new PRINTER_NOTIFY_INFO_DATA[1];
+
+        @Override
+        public void read() {
+            int count = (Integer) readField("Count");
+            aData = new PRINTER_NOTIFY_INFO_DATA[count];
+            if (count == 0) {
+                Count = count;
+                Version = (Integer) readField("Version");
+                Flags = (Integer) readField("Flags");
+            } else {
+                super.read();
+            }
+        }
+
+    }
+
+    /**
+     * A struct containing non-numeric notification data - conditional content
+     * of a {@link NOTIFY_DATA} union.
+     */
+    @Structure.FieldOrder({ "cbBuf", "pBuf" })
+    public class NOTIFY_DATA_DATA extends Structure {
+
+        /**
+         * Indicates the size, in bytes, of the buffer pointed to by pBuf.
+         */
+        public int cbBuf;
+
+        /**
+         * Pointer to a buffer that contains the field's current data.
+         */
+        public WinDef.PVOID pBuf;
```

Comment:
I would bind this as a plain `Pointer` - I don't see the benefit to use a `PVOID` here. It just requires additional un/wrapping

Reply:
Done